source/xep/omemo/mock/purple.c

```diff
@@ -278,4 +293,51 @@ void lurch_topic_update_im(PurpleConversation * conv_p)
 void lurch_topic_update_chat(PurpleConversation * conv_p)
 {}
 
+void purple_conv_chat_write(PurpleConversation * chat_p, char * id, char *msg, enum PurpleFlags flags, time_t time)
+{
+    fprintf(stderr, "%s: %s\n", id, msg);
+}
+
+PurpleConversation *purple_find_conversation_with_account(enum PurpleConvTyp typ, char *to, PurpleAccount *foo)
+{
+    char *room = jabber_get_bare_jid(to);
+    PurpleConversation *return_val = NULL;
+
+    if(purple_rooms_p)
+    {
+        for (GList *it=purple_rooms_p; it; it=it->next)
+        {
+            if(it && it->data)
+            {
+                JabberChat *jc = it->data;
+
+                if(jc && !g_strcmp0(jc->jid, room))
+                {
+                    return_val = it->data;
+                    break;
+                }
+            }
+        }
+    }
+
+    g_free(room);
+
+    return return_val;
+}
```

Comment:
Does this really work? I would assume that this has to be handled by some Shmoose code?

Reply:
Yes it works. I tested it between Conversations and Schmoose.

Reply:
Hm, ok I did not expect that. Until now, I only added a minimum set of code to just get the functionality of purple done which was needed for the lurch interface.